transport/src/main/java/io/netty/channel/socket/oio/OioDatagramChannel.java

```diff
@@ -110,15 +109,21 @@ public OioDatagramChannel(MulticastSocket socket) {
         }
 
         this.socket = socket;
-        config = new DefaultDatagramChannelConfig(this, socket);
+        config = new DefaultOioDatagramChannelConfig(this, socket);
     }
 
     @Override
     public ChannelMetadata metadata() {
         return METADATA;
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * This can be safetly cast to {@link OioDatagramChannelConfig}.
+     */
     @Override
+    // TODO: Change return type to OioDatagramChannelConfig in next major release
```

Comment:
add it to an issue somewhere so we don't forget? Maybe the "TODO next major version" issue?